scripts/tests/test_zkai_native_attention_mlp_rmsnorm_label_policy_gate.py

```
@@ -0,0 +1,140 @@
+    def test_write_outputs_rejects_invalid_payload(self) -> None:
+        payload = gate.build_payload()
+        payload["promotion_policy"]["multi_label_frontier_promotable"] = True
+        gate.refresh_payload_commitment(payload)
+
+        with self.assertRaisesRegex(gate.LabelPolicyError, "multi-label frontier"):
+            gate.write_outputs(payload, gate.JSON_OUT, None)
+
+    def test_write_outputs_rejects_outside_evidence_dir(self) -> None:
+        payload = gate.build_payload()
+        with tempfile.TemporaryDirectory() as tmpdir:
+            with self.assertRaises(gate.LabelPolicyError):
```

Ruff (0.15.12)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)